src/global.d.ts

```diff
@@ -0,0 +1,26 @@
+
```

Comment:
Can you add these types to `index.ts` instead - the reason is not a lot of tooling supports `.d.ts` files since they are supposed to be machine generated so it's easy for typos and mistakes to creep in that we only find out about after publishing.

Reply:
should we just use the types from fails-components/webtransport?

Reply:
> the reason is not a lot of tooling supports .d.ts files since they are supposed to be machine generated so it's easy for typos and mistakes to creep in that we only find out about after publishing.

.d.ts types aren't only supposed to be machine generated, and typescript (aegir build) will fail if any of the types in .d.ts are broken.  

However, conflicting global namespace definitions could be an issue. I've updated this PR to use `@fails-component/webtransport` types, and defining WebTransport on the global namespace via `types.ts`

Reply:
> .d.ts types aren't only supposed to be machine generated, and typescript (aegir build) will fail if any of the types in .d.ts are broken.

We've definitely had loads of trouble with hand-rolled .d.ts files in the past, so I don't think this is enough - just search on the js-ipfs repo and you'll see lots of issues.  Not really keen on going down that route again.

Reply:
Sounds like build tools werent set up properly (people like to enable skipLibCheck) but i’ll take your word for it!